### PR TITLE
feat: add textDecorationLine support for typography

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
Fixes: https://github.com/razorpay/blade/issues/1330

Co-authored-by: archie252000 <arc252000@gmail.com>

### CODE REVIEW

File: .changeset/eight-emus-arrive.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@razorpay/blade": patch
+---
+
+feat: add text decoration support for typography
```
Comment (anuraghazra) [inline .changeset/eight-emus-arrive.md:5]: two changelogs?

Comment (chaitanyadeorukhkar) [reply .changeset/eight-emus-arrive.md]: Woops

**Review by anuraghazra — APPROVED** [review]
Mostly looks good, let's add tests if required.

File: packages/blade/src/components/Typography/Title/Title.tsx [hunk 2]
```diff
@@ -82,14 +83,20 @@ export const Title = ({
...
   useValidateAsProp({ componentName: 'Title', as, validAsValues });
 
   const props = getProps({ as, size, type, contrast, color, testID });
 
   return (
-    <BaseText {...props} textAlign={textAlign} {...getStyledProps(styledProps)}>
+    <BaseText
+      {...props}
+      textAlign={textAlign}
+      textDecorationLine={textDecorationLine}
```
Comment (anuraghazra) [inline packages/blade/src/components/Typography/Title/Title.tsx:97]: Also supported on react native right?

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Typography/Title/Title.tsx:97]: Yes sir! https://reactnative.dev/docs/text-style-props#textdecorationline

Comment (anuraghazra) [reply packages/blade/src/components/Typography/Title/Title.tsx:97]: Noice.

**Pull request merged**